### PR TITLE
support medata filter in the REST API

### PR DESCRIPTION
**Pull request by pmeier**

### CODE REVIEW

File: ragna/core/_rag.py [hunk 1]
```diff
@@ -191,7 +191,7 @@ async def prepare(self) -> Message:
                 "Chat is already prepared",
                 chat=self,
                 http_status_code=400,
-                detail=RagnaException.EVENT,
+                http_detail=RagnaException.EVENT,
```
Comment (pmeier) [inline ragna/core/_rag.py:194]: Driveby because I hit this during development and I was wondering why I don't get diagnostics back.

File: ragna/deploy/_api/orm.py [hunk 2]
```diff
@@ -19,14 +19,16 @@ class Json(types.TypeDecorator):
 
     cache_ok = True
 
-    def process_bind_param(self, value: Any, dialect: Dialect) -> str:
+    def process_bind_param(self, value: Any, dialect: Dialect) -> Optional[str]:
+        if value is None:
+            return value
```
Comment (pmeier) [inline ragna/deploy/_api/orm.py:24]: We now also have a nullable JSON field with the `MetadataFilter` and thus also need to handle this here.

File: ragna/deploy/_api/orm.py [hunk 3]
```diff
@@ -72,12 +74,27 @@ class Document(Base):
...
+class MetadataFilter(Base):
+    __tablename__ = "metadata_filters"
+
+    id = Column(types.Uuid, primary_key=True)  # type: ignore[attr-defined]
+    user_id = Column(ForeignKey("users.id"))
+
+    data = Column(Json, nullable=False)
+
+    chat_id = Column(ForeignKey("chats.id"), nullable=True)
+    chat = relationship("Chat", back_populates="metadata_filter", uselist=False)
```
Comment (pmeier) [inline ragna/deploy/_api/orm.py:86]: 1-to-1 relationship with a chat

**Review by nenb — APPROVED** [review]
Approving to unblock @pierrotsmnrd.

Will do a post-merge review now.

**Comment by nenb** [comment]
_Taken from Slack:_

It seems fine. The only caveat that I have is that I may have missed something when it comes to serialising the metadata filter to the database, but I think the PR is small enough that it's fine!
